site/src/components/DropdownButton/DropdownButton.tsx

```diff
@@ -0,0 +1,113 @@
+import Button from "@material-ui/core/Button"
+import Popover from "@material-ui/core/Popover"
+import { makeStyles } from "@material-ui/core/styles"
+import { CloseDropdown, OpenDropdown } from "components/DropdownArrows/DropdownArrows"
+import { DropdownContent } from "components/DropdownButton/DropdownContent/DropdownContent"
+import { FC, ReactNode, useEffect, useRef, useState } from "react"
+import { CancelButton } from "./ActionCtas"
+
+export interface DropdownButtonProps {
+  primaryAction: ReactNode
+  secondaryActions: Array<{ action: string; button: ReactNode }>
+  canCancel: boolean
+  handleCancel?: () => void
+}
+
+export const DropdownButton: FC<DropdownButtonProps> = ({
+  primaryAction,
+  secondaryActions,
+  canCancel,
+  handleCancel,
+}) => {
+  const styles = useStyles()
+  const anchorRef = useRef<HTMLButtonElement>(null)
+  const [isOpen, setIsOpen] = useState(false)
+  const id = isOpen ? "action-popover" : undefined
+
+  /**
+   * Ensures we close the popover before calling any action handler
+   */
+  useEffect(() => {
```

Comment:
What is the use case for this?

Reply:
Good catch, I meant to come back to this. It used to take a dependency that's no longer available, but I don't think it's the best way to handle it anyway. I switched to closing the popover on blur.